src/java.base/share/classes/java/lang/RuntimeException.java

```diff
@@ -117,4 +117,27 @@ protected RuntimeException(String message, Throwable cause,
                                boolean writableStackTrace) {
         super(message, cause, enableSuppression, writableStackTrace);
     }
+
+    /**
+     * Throws a new RuntimeException if the given parameter
+     * evaluates to true. Otherwise, the method exits with no action.
+     *
+     * This allows for concise checking of preconditions at the start
+     * of a method. Also, if a child class of RuntimeException calls
+     * this method, then that child class will be the exception that
+     * is thrown rather than RuntimeException itself. For example,
+     * IllegalArgumentException.throwIf(3 > 2) will throw an
```

Comment:
`IllegalArgumentException.throwIf(3 > 2)`
Uh, this method you added is an instance method, and you can't call it like that.

In general, you should avoid creating exception instances if you don't need to throw them, since filling in stack trace is extremely costly.

Reply:
Terribly sorry, I meant to do it like this.

    new IllegalArgumentException.throwIf(3 > 2);

Thank you for pointing that out.

But yes, your point still stands - it is expensive to create stack traces, so therefore, this is not an "equivalent, but more concise" solution, it is significantly more costly in exchange for some readability.

Perhaps it actually would work better as a static method? Then, the stack trace is only created upon breaking the precondition.

Thank you again for your feedback.

Reply:
Something like this.

```
public static void throwIf(boolean throwGuard, String message) {
    if (throwGuard) {
        throw new RuntimeException(message);
    }
}
```

Reply:
I recommend you learn the Java Programming Language, such as how static methods work, before making contributions to the OpenJDK. See section 5 of https://openjdk.org/contribute/

Reply:
> I recommend you learn the Java Programming Language, such as how static methods work, before making contributions to the OpenJDK. See section 5 of https://openjdk.org/contribute/

I believe the PR author really means something like
```
IllegalArgumentException exception = new IllegalArgumentException();
exception.throwIf(...);
```
where throwIf is called on an instance of an exception, instead of a check in an if block as per usual. Whether this is a good solution compared to the usual approach I can't comment on, but I think there might be a bit of misunderstanding involved

Reply:
@TheShermanTanker, I believe the reason why they made that comment was because my static function suggestion does not satisfy the goals I originally claimed this pull request would satisfy.

Looking at my static function suggestion, the method is no longer specialized for each Exception class. Yes, I can do `IllegalArgumentException.throwIf(true)`, but I will throw a RuntimeException, not an IllegalArgumentException.

I did not clarify that I was moving the goal posts, so I can understand if it came off like I thought a static method would somehow accomplish the original goals that I had set for myself. I think that that is why they suggested that I learn the language and how static methods work within it.

@liach, please feel free to correct me if I am misrepresenting you.

Reply:
Indeed, static methods won't work and you understand the reason.

I suggest you look at a Preconditions class in jdk.internal.util package for how jdk handles common exceptions. I suggest writing personal utilities or using library utilities if there is a particular type of IAE that happens often to you than adding this somewhat useless method, only good for it squashes code to one possibly too long line.

Reply:
Thank you for responding and clarifying @liach. Your patience is appreciated.

I believe I have found the file you are suggesting. I attempted to use this class in my code, but it does not work. Likely because it is an internal jdk package. Regardless, this gives a template on how to create Precondition checks for myself. 

https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/jdk/internal/util/Preconditions.java

This definitely makes sense. By using a java.util.function.* class, I can defer object creation (or in this case, exception creation) until the moment that I use it, allowing me to bypass the concern you raised before. Alternatively, I could just create a custom method that does what I need it to, which segues into your point.

Based off what you are saying, my feature is something that doesn't provide enough value to warrant its inclusion. I understand what you are saying and it makes sense. Based off of that, it likely would be best for me to close this pull request and focus on making something else.

But to get a feel of the mindset of JDK Contributors, I'd like to ask a hypothetical question. If exception creation was not expensive, would this feature make sense to include in the JDK? I want to understand the level of quality I must aim for before making my next pull request to the JDK.

Reply:
> _But to get a feel of the mindset of JDK Contributors, I'd like to ask a hypothetical question. If exception creation was not expensive, would this feature make sense to include in the JDK? I want to understand the level of quality I must aim for before making my next pull request to the JDK._

@TheShermanTanker and @kristylee88, I'd like to direct this question to you 2 too. I am new and learning, and I would like to come out of this with as much knowledge as I can about how to do better. If not that, then any other suggestions you all have to give would be greatly appreciated.